mantidproject/mantid
# Adding automatic creation of all group/pairs for a workspace in muon analysis

The muon group wanted checkboxes, which when selected would automatically create the workspaces for all of the groups/pairs. In the current release (3.11) the workspaces are only generated when they are needed (plotted or used in multiple fitting). 

**To test:**
1. Turn on the data archieve
2. Open Muon analysis and select MUSR as the instrument.
3. Enter 62260 as the run number (some data should then load)
4. Go to the grouping tab and create a group and two pairs (add them to the tables)
5. Go to the settings tab
6. Expand the grouped workspace MUSR00062260 and it will contain 2 workspaces (binned and raw data)
7. Tick the 'Load all groups" option. The grouped workspace will now have 10 workspaces (5 binned and 5 raw).
8. Go to the grouping tab and add another group. It should be added to the grouped workspace.
9. Go to the settings tab and untick Load all groups
10. Tick Laod all pairs. The 2 pairs should be added to the grouped workspace.
11. In the grouping tab create another pair and it should be added to the grouped workspace.
12. In the settings tab tick Load all groups
13. In the home tab enter MUSR00067063 into the load run bar.
14. A grouped workspace called MUSR00067063 will be created and will contain all of the groups and pairs.

Fixes #21397. 

**Release Notes** 

Added to release notes and muon Analysis documentation has been updated. 
---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
